ga/client/rttestimator.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2016 Chun-Ying Huang
+ *
+ * This file is part of GamingAnywhere (GA).
+ *
+ * GA is free software; you can redistribute it and/or modify it
+ * under the terms of the 3-clause BSD License as published by the
+ * Free Software Foundation: http://directory.fsf.org/wiki/License:BSD_3Clause
+ *
+ * GA is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * You should have received a copy of the 3-clause BSD License along with GA;
+ * if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
+
+#ifdef _WIN32
+#include <winsock2.h>
+#include <Ws2tcpip.h>
+#else
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <unistd.h>
+#endif
+
+#include "ga-common.h"
+#include "controller.h"
+#include "rttestimator.h"
+
+#define PORT 8556
+#define BUFSIZE 512
+
+static pthread_mutex_t rtt_mutex;
+static unsigned int rtt_store[RTT_STORE_SIZE]; // RTT values in microseconds
+static unsigned int ping_id_iterator;
```

Comment:
This needs to be refactored. Currently, it serves no purpose and actually breaks the code.